File: backend/src/main/java/com/example/backend/Exercise.java
package com.example.backend.Exercise;

import jakarta.persistence.*;

public class Exercise {

    private Long id;
    private String name;
    private String type;
    private String muscle;
    private String difficulty;
    private String instructions;

    public Exercise(String name, String type, String muscle, String difficulty, String instructions) {
        this.name = name;
        this.type = type;
        this.muscle = muscle;
        this.difficulty = difficulty;
        this.instructions = instructions;
    }

    public Exercise() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getMuscle() {
        return muscle;
    }

    public void setMuscle(String muscle) {
        this.muscle = muscle;
    }

    public String getDifficulty() {
        return difficulty;
    }

    public void setDifficulty(String difficulty) {
        this.difficulty = difficulty;
    }

    public String getInstructions() {
        return instructions;
    }

    public void setInstructions(String instructions) {
        this.instructions = instructions;
    }
}

File: backend/src/main/java/com/example/backend/exercise_generator/model/Exercise.kt
package com.example.backend.exercise_generator.model

import com.example.backend.user.model.Program
import jakarta.persistence.*

@Entity
class Exercise {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0
    var name: String = ""
    var type: String = ""
    var muscle: String? = null
    var equipment: String? = null
    var difficulty: String? = null
    var instructions: String? = null

    @ManyToOne
    @JoinColumn(name = "program_id")
    var program: Program? = null
}